### PR TITLE
Feature/split network config

### LINKED ISSUE #1993: Review network config
- Decouple network config (addresses, genesis, ...) => should be constant per deployment VS software config (sequencer params, ...)
- Enable default networks `-- network mainnet | testnet`

### PR DESCRIPTION
**Pull request by arnaubennassar**
Closes #1993

### What does this PR do?

- Move network related configs from config.toml => genesis.json
- Add flag option to load default networks without the need of a genesis file (for mainnet and testnet)
- Update default config
- Rename outdated PoE to ZkEVM

### CODE REVIEW

File: config/environments/local/local.genesis.config.json [hunk 1]
```diff
@@ -1,5 +1,12 @@
 {
+  "L1Config" : {
+		"L1ChainID": 1337,
+		"PoEAddr": "0x610178dA211FEF7D417bC0e6FeD39F05609AD788",
+		"MaticAddr": "0x5FbDB2315678afecb367f032d93F642f64180aa3",
+		"GlobalExitRootManagerAddr": "0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6"
```
Comment (ARR552) [inline config/environments/local/local.genesis.config.json:6]: I think this is a good oportunity to change this names to:
```suggestion
		"L1ChainID": 1337,
		"PolygonZkEVMAddress": "0x610178dA211FEF7D417bC0e6FeD39F05609AD788",
		"MaticAddress": "0x5FbDB2315678afecb367f032d93F642f64180aa3",
		"PolygonZkEVMGlobalExitRootAddress": "0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6"
```
I would be good to ask the protocol team to change the format of the json to include this parameters automatically instead having to create it manually

File: config/network.go [hunk 2]
```diff
@@ -14,15 +17,24 @@ import (
...
 // NetworkConfig is the configuration struct for the different environments
 type NetworkConfig struct {
+	L1Config                    etherman.L1Config
 	L2GlobalExitRootManagerAddr common.Address
 	L2BridgeAddr                common.Address
 	Genesis                     state.Genesis
 	MaxCumulativeGasUsed        uint64
 }
 
+type supportedNetworks string
```
Comment (ARR552) [inline config/network.go:27]:
```suggestion
type network string
```

File: docker-compose.yml [hunk 3]
```diff
@@ -27,11 +27,10 @@ services:
       - ZKEVM_NODE_ETHERMAN_URL=${ZKEVM_NODE_ETHERMAN_URL}
     volumes:
       - ${ZKEVM_ADVANCED_CONFIG_DIR:-./config/environments/public}/public.node.config.toml:/app/config.toml
-      - ${ZKEVM_ADVANCED_CONFIG_DIR:-./config/environments/public}/public.genesis.config.json:/app/genesis.json
     command:
       - "/bin/sh"
       - "-c"
-      - "/app/zkevm-node run --genesis /app/genesis.json --cfg /app/config.toml --components rpc"
+      - "/app/zkevm-node run --network ${ZKEVM_NETWORK} --cfg /app/config.toml --components rpc"
```
Comment (ARR552) [inline docker-compose.yml:33]: Why do we need this change?? I don't use env so I think i will have to modify this file all the time

File: config/mainnetgenesis.go [hunk 4]
```diff
@@ -1,5 +1,16 @@
+package config
```
Comment (ARR552) [inline config/mainnetgenesis.go:1]: Why don't we keep this file under the mainnet directory? (same for testnet genesis)

Comment (arnaubennassar) [reply config/mainnetgenesis.go:1]: Not a bad idea... But not sure either 🤔 for me this is like the default node config, that also lives in this package

**Review by ARR552 — APPROVED** [review]

**Pull request merged**